### PR TITLE
Update [ThreadStatic] naming violations

### PR DESCRIPTION
**Pull request by hughbe**
From @danmosensft in #20898

> I see 27 in the sources and some need changing from s_ to t_

Here's me doing that

### CODE REVIEW

File: src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/TaskAwaiterTests.cs [hunk 1]
```diff
@@ -330,7 +330,7 @@ public static IEnumerable<object[]> CanceledTasksAndExpectedCancellationExceptio
         private class ValidateCorrectContextSynchronizationContext : SynchronizationContext
         {
             [ThreadStatic]
-            internal static bool IsPostedInContext;
+            internal static bool s_isPostedInContext;
```
Comment (danmosemsft) [inline src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/TaskAwaiterTests.cs]: t_ ?

File: src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/YieldAwaitableTests.cs [hunk 2]
```diff
@@ -150,7 +150,7 @@ public static void RunAsyncYieldAwaiterTests_Negative()
         private class ValidateCorrectContextSynchronizationContext : SynchronizationContext
         {
             [ThreadStatic]
-            internal static bool IsPostedInContext;
+            internal static bool s_isPostedInContext;
```
Comment (danmosemsft) [inline src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/YieldAwaitableTests.cs]: t_ ?

File: src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/TaskAwaiterTests.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/TaskAwaiterTests.cs]: :( dammit

**Comment by hughbe** [comment]
Updated the names - that's what you get for doing this first thing in the morning!

**Comment by hughbe** [comment]
Test Innerloop OSX10.12 Release x64 Build and Test

**Comment by danmosemsft** [comment]
Thanks Hugh.

**Pull request merged**